### PR TITLE
translate and check in design docs

### LINKED ISSUE #738: get design docs on github
we still have @spencerkimball's [original design docs](https://docs.google.com/document/d/11k2EmhLGSbViBvi6_zFEiKzuXxYF49ZuuDJLe6O8gBU/edit#heading=h.kgqgd63x1fet) online. They should go to our github page and ideally make it easy for others to contribute to them as well (I have found it tedious to edit such a large document WYSIWYG).

There are two options:

- the github wiki, but that doesn't allow pull requests.
- just a folder in our repo with a README.md entry point. [Demo](https://github.com/tschottdorf/cockroach/tree/tschottdorf/design_docs/docs)

I like the second option more because it's more community-friendly and you can browse and read it just fine, so it works pretty much like the wiki plus pull requests. With either option it'll be easy to migrate it later. In any case, now's a good time to do this because we've been discussing changes that would benefit from being put into writing.

### PR DESCRIPTION
**Pull request by tbg**
certainly not a perfect job, but I hope I've managed to get
the bulk of the tiring editing done.

fixes #738

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -0,0 +1,1319 @@
...
+[here](https://docs.google.com/document/d/1kBCu4sdGAnvLqpT-_2vaTbomNmX3_saayWEGYu1j7mQ/edit?usp=sharing).
+
+**Transaction Table**
+
+```
+enum Isolation {
+
+SNAPSHOT = 1;
+
+SERIALIZABLE = 2;
```
Comment (petermattis) [inline docs/design.md]: Can you indent these and remove the extra blank lines?

File: docs/design.md [hunk 2]
```diff
@@ -0,0 +1,1319 @@
...
+ SSD = 1;
+ HDD = 2;
+ MEM = 3;
+ };
+ optional DiskType disk_type = 2;
+ } replicas = 1;
+ // Minimum and maximum range sizes to control splits / merges.
+ optional uint32 range_min_size = 2;
+ optional uint32 range_max_size = 3;
+}
```
Comment (petermattis) [inline docs/design.md]: A little bit better indentation will make this more readable.

File: docs/design.md [hunk 3]
```diff
@@ -0,0 +1,1319 @@
...
+`\0perm<key-prefix><user>`
+
+Permission values are a protobuf containing the permission details:
+
+```
+UserPermission {
+ bool read = 1 [default = True];
+ bool write = 2 [default = True];
+ float priority = 3 [default = 1.0];
+}
```
Comment (petermattis) [inline docs/design.md]: Indentation.

**Comment by petermattis** [comment]
LGTM

**Comment by spencerkimball** [comment]
LGTM

File: docs/design.md (hunk 1 above)
Comment (tschottdorf) [inline docs/design.md]: I've actually removed it - it's quite out of date. We'll have to figure out how to best do this, but for now I'll link directly to the proto with a hint on what to read.

File: docs/design.md (hunk 2 above)
Comment (tschottdorf) [inline docs/design.md]: removed and linked to proto.

File: docs/design.md (hunk 3 above)
Comment (tschottdorf) [inline docs/design.md]: ditto.

**Pull request merged**